### PR TITLE
[Lake][Delete 2/2] Support read and compact lake tablet with delete rowset

### PR DESCRIPTION
**Pull request by wyb**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

1. Add delete predicate processing in lake tablet reader and compaction policy.
2. Fix update cumulative point bug in base compaction.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by sduzh — DISMISSED** [review]

**Review by sduzh — APPROVED** [review]

File: be/src/storage/lake/tablet_reader.cpp [hunk 1]
```diff
@@ -121,6 +133,78 @@ Status TabletReader::init_predicates(const TabletReaderParams& params) {
     return Status::OK();
 }
 
+Status TabletReader::init_delete_predicates(const TabletReaderParams& params, DeletePredicates* dels) {
```
Comment (chaoyli) [inline be/src/storage/lake/tablet_reader.cpp:136]: If you reimplement the interface, can we reuse the predicates of query engine?

File: be/src/storage/lake/rowset.cpp [hunk 2]
```diff
@@ -39,6 +38,9 @@ StatusOr<ChunkIteratorPtr> Rowset::read(const vectorized::Schema& schema, const
     seg_options.chunk_size = options.chunk_size;
     seg_options.global_dictmaps = options.global_dictmaps;
     seg_options.unused_output_column_ids = options.unused_output_column_ids;
+    if (options.delete_predicates != nullptr) {
```
Comment (chaoyli) [inline be/src/storage/lake/rowset.cpp:41]: delete predicate is not stored in meta? Why to user _index?

File: be/src/storage/lake/tablet_reader.cpp (hunk 1 above)
Comment (wyb) [reply be/src/storage/lake/tablet_reader.cpp:136]: Yes, it is for lake tablet.

File: be/src/storage/lake/rowset.cpp (hunk 2 above)
Comment (wyb) [reply be/src/storage/lake/rowset.cpp:41]: The index is used to get the delete predicate which version > this rowset.

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wyb** [comment]
run starrocks_admit_test

**Pull request merged**